src/main/java/frc/robot/subsystems/GamePieceManipulatorSubsystem.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+package frc.robot.subsystems;
+
+import static frc.robot.Constants.GamePieceManipulatorConstants.DEVICE_ID_CANrange;
+import static frc.robot.Constants.GamePieceManipulatorConstants.DEVICE_ID_MANIPULATORMOTOR;
+import static frc.robot.Constants.GamePieceManipulatorConstants.IntakeSpeed;
+import static frc.robot.Constants.GamePieceManipulatorConstants.OuttakeSpeed;
+import static frc.robot.Constants.GamePieceManipulatorConstants.ScoreSpeed;
+
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.CANrange;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.units.measure.Distance;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class GamePieceManipulatorSubsystem extends SubsystemBase {
+  /** Creates a new ExampleSubsystem. */
+
+  // define motors
+  public final TalonFX manipulatorMotor = new TalonFX(DEVICE_ID_MANIPULATORMOTOR);
+
+  CANrange canrange = new CANrange(DEVICE_ID_CANrange);
+  // Configure the bacic use
+  // CANrangeConfiguration configs = new CANrangeConfiguration();
+
+  // write these configs to CANrange
+  // canrange.getConfigurator().apply(configs);
+
+  StatusSignal<Distance> distance = canrange.getDistance();
+
+  // Convert distance in to a float so we can use it
+  String distanceString = distance.refresh().toString();
+  float convertedDistance = Float.parseFloat(distanceString);
+
+  // define how motors are controlled
+  private final VelocityTorqueCurrentFOC wheelControl = new VelocityTorqueCurrentFOC(0);
+
+  public GamePieceManipulatorSubsystem() {
+  }
+
+  public void intakeCoral() {
+    manipulatorMotor.setControl(wheelControl.withVelocity(IntakeSpeed));
+  }
+
+  public void outtakeCoral() {
+    manipulatorMotor.setControl(wheelControl.withVelocity(OuttakeSpeed));
+  }
+
+  public void scoreCoral() {
+    manipulatorMotor.setControl(wheelControl.withVelocity(ScoreSpeed));
+  }
+
+  public void stop() {
+    manipulatorMotor.stopMotor();
+  }
+
+  public void update() {
+    if (convertedDistance > 20) {
+      stop();
+    }
+  }
```

Comment:
I don't think there should be an `update()` method that stops the mechanism in the subsystem. It would make sense to have a more abstract concept here, like a method to say if a game piece is in position, then the command will deal with what to do.